textile/features.textile

```diff
@@ -525,9 +525,9 @@ h3(#realtime-connection). Connection
 * @(RTN10)@ This clause has been deleted. It was valid up to and including specification version @1.2@.
 * @(RTN11)@ @Connection#connect@ function:
 ** @(RTN11a)@ Explicitly connects to the Ably service if not already connected
-** @(RTN11b)@ If the state is @CLOSING@, the client should make a new connection with a new transport instance and remove all references to the old one. In particular, it should make sure that, when the @CLOSED@ @ProtocolMessage@ arrives for the old connection, it doesn't affect the new one.
+** @(RTN11b)@ If the state is @CLOSING@, the client should make a new connection with a new transport instance and remove all references to the old one. In particular, it should make sure that, when the @CLOSED@ @ProtocolMessage@ arrives for the old connection, it doesn't affect the new one. Additionally, the client should ensure that all channels first transition to @DETACHED@, following "@RTL3b@":#RTL3b, and then reinitialize channels per "@RTN11d@":#RTN11d.
```

Comment:
```suggestion
** @(RTN11b)@ If the state is @CLOSING@, the client should make a new connection with a new transport instance and remove all references to the old one. In particular, it should make sure that, when the @CLOSED@ @ProtocolMessage@ arrives for the old connection, it doesn't affect the new one and then reinitialize channels per "@RTN11d@":#RTN11d.
```
Not sure if we need to make sure channels go detached. If we are clearing `recoveryKey` and resetting `msgSerial` then there shouldn't be issue about detaching channels.
The latest ably-java PR https://github.com/ably/ably-java/pull/1077/files, doesn't seem to detach channels.

Reply:
It detaches channels, there are even tests for that. The reason why I added this is because state change from attached to initialized is weird and never mentioned in the spec, we should either rework the spec and ensure we cover this branch (and now we don't cover at least for presence, we should definitely clear the presence if we closed client and reconnected) or make it explicit detach before reconnection

Reply:
Yeah, seems like we do channel detach when connection close message is received